src/main/java/org/myrobotlab/service/ServoMixer.java

```diff
@@ -41,6 +42,11 @@ public ServoMixer(String n, String id) {
     for (String sc : all) {
       allServos.add(Runtime.getFullName(sc));
     }
+    
+    File poseDirectory = new File(posesDirectory);

```

Comment:
shouldn't this be done in startService, not in the constructor?

Reply:
I don't think so ... 
Some services have a concept of "default" and we should try to provide this as best as possible ..
I found the service exploding previously, because a call to get the list of files in the "default" directory was null

Reply:
This means that you have the side effect of creating a directory on the filesystem, and you have no control over changing that default location.  Initialization stuff should happen in a startService method.  The creating of the directory shouldn't happen in the constructor of this object.